athena_cli.py

```diff
@@ -9,25 +10,69 @@
 import sys
 import time
 import uuid
+import itertools
 
 import boto3
 import botocore
 import cmd2 as cmd
 from botocore.exceptions import ClientError, ParamValidationError
 from tabulate import tabulate
 
-LESS = "less -FXRSn"
+LESS = "less -FXRn"
+LESS_TRUNC = "less -FXRSn"
 HISTORY_FILE_SIZE = 500
 
-__version__ = '0.1.8'
+__version__ = '0.2.0'
+
+def output_results(athena, format, execution_id, output, is_shell):
+    results = athena.get_query_results(execution_id)
+    headers = results[0]
+    counter = itertools.count(start=1)
+    rows = zip(results[1], counter)
+
+    try:
+        if format in ['CSV', 'CSV_HEADER', 'TSV', 'TSV_HEADER']:
+            if format in ['TSV', 'TSV_HEADER']:
+                delim = '\t'
+                quote = csv.QUOTE_NONE
+                esc = '\\'
+            else:
+                delim = ','
+                quote = csv.QUOTE_ALL
+                esc = None
+
+            csv_writer = csv.writer(output, delimiter=delim, quoting=quote, escapechar=esc)
+
+            if format in ['CSV_HEADER', 'TSV_HEADER']:
+                csv_writer.writerow(headers)
+            csv_writer.writerows([row for row, count in rows])
+
+        elif format == 'VERTICAL':
+            for row, count in rows:
+                output.write('--[RECORD {}]--'.format(count))
+                output.write('\n')
+                output.write(tabulate(zip(*[headers, row]), tablefmt='presto'))
+                output.write('\n')
+
+        else:  # ALIGNED
+            output.write(tabulate([row for row, count in rows], headers=headers, tablefmt='presto'))
+            output.write('\n')
+
+        output.flush()
+    except IOError as x:
+        # quitting the less process in shell causes an IOError, so ignore
+        if not is_shell:
+            raise x
+
+    return next(counter) - 1
 
 
 class AthenaBatch(object):
 
-    def __init__(self, athena, db=None, format='CSV'):
+    def __init__(self, athena, db=None, format=None):
         self.athena = athena
         self.dbname = db
-        self.format = format
+        self.format = 'CSV' if format is None else format
```

Comment:
A more idiomatic way of writing this is ...
```
self.format = format or 'CSV'
```
However, why did you remove the default format from the method signature?

Reply:
The default wasn't being used anymore, it is always set, so it felt redundant having the default

Reply:
I updated the code as you suggested...